x/profile/internal/types/msgs.go

```diff
@@ -0,0 +1,206 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// ----------------------
+// --- MsgCreateProfile
+// ----------------------
+
+// MsgCreateProfile defines a CreateProfile message
+type MsgCreateProfile struct {
+	Name     string         `json:"name,omitempty"`
+	Surname  string         `json:"surname,omitempty"`
+	Moniker  string         `json:"moniker"`
+	Bio      string         `json:"bio,omitempty"`
+	Pictures *Pictures      `json:"pictures,omitempty"`
+	Creator  sdk.AccAddress `json:"creator"`
+}
+
+// NewMsgCreateProfile is a constructor function for MsgCreateProfile
+func NewMsgCreateProfile(name string, surname string, moniker string, bio string, pictures *Pictures,
+	creator sdk.AccAddress) MsgCreateProfile {
+	return MsgCreateProfile{
+		Name:     name,
+		Surname:  surname,
+		Moniker:  moniker,
+		Bio:      bio,
+		Pictures: pictures,
+		Creator:  creator,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgCreateProfile) Route() string { return RouterKey }
+
+// Type should return the action
+func (msg MsgCreateProfile) Type() string { return ActionCreateProfile }
+
+// ValidateBasic runs stateless checks on the message
+func (msg MsgCreateProfile) ValidateBasic() error {
+	if msg.Creator.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("Invalid creator address: %s", msg.Creator))
+	}
+
+	if len(msg.Name) > MaxNameSurnameLength {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, fmt.Sprintf("Profile name cannot exceed %d characters", MaxNameSurnameLength))
+	}
+
+	if len(msg.Surname) > MaxNameSurnameLength {
```

Comment:
Probably we should also have a minimum name and surname length (2 characters, for chinese names) here.